### ISSUE TITLE
Support editing existing comments instead of overwriting

### ISSUE THREAD

**Issue by deecewan** [danger/danger-js#166]
I think it'd be nice (maybe it happens on `danger.rb`?) for problems that are resolved to be struck out? It makes it hard to track progress and reasons for certain commits.

In this, I'd also like to see some logic to check if the comment if actually a danger comment before overwriting.  We use the same bot on a bunch of repos, and it comments about various things in various places.  It took me a while to realise that it was still working, but danger was overwriting the comment instead of creating a new one.

**Comment by orta** [comment]
We do support strikeouts in Danger.rb - I'm hesitant to add it in, as it adds a lot of complexity but ended up rarely used when I asked people if they used it.

In rb there is a `--danger_id` flag that you can pass, and then it would only update/edit/delete the comment with the same ID. This allowed for multiple Danger's to comment on the same thread. Would that fix it?

**Comment by deecewan** [comment]
yeah, as i was writing the strikeout thing, I don't really think it adds much to be honest.

Editing comments would be 100% great, although I'm not sure how I'd know which comment to flag ahead of time?

**Comment by orta** [comment]
Danger JS is already editing comments, and deleting them as they become empty. 

Markdown allows for arbitrary text to be inserted inside HTML elements ;)

<img width="683" alt="screen shot 2017-03-15 at 08 10 25" src="https://cloud.githubusercontent.com/assets/49038/23926321/dfb73900-0956-11e7-9a4d-f4e3a16b610e.png">

So I usually put data like that inside the table element

**Comment by deecewan** [comment]
i guess the thing is how does it know which comment to edit?

I had a case where the bot would comment with the IP of a server, and then danger would run and overwrite the server IP.  After that it just edited it's own comment.

**Comment by orta** [comment]
So the danger-id thing is used for when you have `danger` being ran multiple times on the same commit ( maybe from different build machines etc )

in RB, the flag `--dangerID=123` would mean that inside the markdown there would be a `data-danger-id="123"` then later as Danger is looking for the right comment to edit, it checks that there is a corresponding `dangerID` and the one that was ran during the argument. If it doesn't find one, then it creates a new comment instead

**Comment by deecewan** [comment]
right.  I think what i'm proposing is more 'only edit a comment that Danger has created, otherwise add a new one'.  Maybe we can use a flag for that to retain existing behavior if people want it.

**Comment by orta** [comment]
This feels like the current behavior to me though? can you write out a flow
of what would happen after these changes

**Comment by orta** [comment]
Related #196

**Comment by deecewan** [comment]
yes! this is pretty much exactly what I was meaning, i think.

**Comment by pranshuchittora** [comment]
Hi, is there any workaround/hotfix for the same. This feature is really required in our monorepo setup with multiple danger files. 
Thank you : )
